src/test/scala/it/unibo/authsim/library/user/UserBuilderTest.scala

```diff
@@ -20,19 +20,19 @@ class UserBuilderTest extends AnyWordSpec with should.Matchers{
   private val password: String= "password1"
 
   private val costumUserBuilder1 = UserCostumBuilder() withName(name) withPassword(password)
-  private val costumUser1 :Option[User]= costumUserBuilder1.build()
+  private val costumUser1 :Option[User]= costumUserBuilder1.build
 
   private val costumUserBuilder2 =  UserCostumBuilder() withName(name) withPassword(password) withPolicy(userIDPolicy)
-  private val costumUser2:Option[User] = costumUserBuilder2.build()
+  private val costumUser2:Option[User] = costumUserBuilder2.build
 
   private val costumUserBuilder3 =  UserCostumBuilder() withName(shortName) withPassword(password) withPolicy(userIDPolicy)
-  private val costumUser3:Option[User] = costumUserBuilder3.build()
+  private val costumUser3:Option[User] = costumUserBuilder3.build
 
   private val autoUserBuilder1= UserAutoBuilder() withPolicy(userIDPolicy)
```

Comment:
@asperan have you try to add `UserAutoBuilder` as ` autoUserBuilder1` type to remove `asInstanceOf`?
I mean as it follows:
`  private val autoUserBuilder1: UserAutoBuilder= UserAutoBuilder() withPolicy(userIDPolicy)`
`  private val autoUser1:User = autoUserBuilder1.build`

Reply:
You are right, and we can leave type inference to Scala, so we can also leave out the type specification:

```
private val autoUserBuilder1= UserAutoBuilder() withPolicy(userIDPolicy)
private val autoUser1:User = autoUserBuilder1.build
```

I'll do a commit with the changes.